Nutch-2676

sbatururimi opened the pull request:
This is a patch containing some updates to the selenium plugin:
- headless mode for Chrome/Firefox
- code for Selenium HUB 
- customers rotation to WebDriver in remote mode
- examples of use with Docker (specifically with Docker Compose) as well with Docker Selenium nodes, Docker Selenium Hub, Docker Apache Solr 
- deprecated code removed

sebastian-nagel commented on lib/README.txt:
```diff
@@ -0,0 +1,45 @@
+This is a placeholder to put Hadoop native libraries -it's a component
```
This file is duplicated from lib/native/README.txt - is there a reason?

sbatururimi replied:
Not any. Inadvertently added

sbatururimi commented on src/plugin/lib-selenium/README.md:
```diff
@@ -0,0 +1,13 @@
+* Headless mode has been added for Chrome and Firefox. Set `selenium.enable.headless` to `true` in nutch-default.xml or nutch-site.xml to use it.
+
+
+Your can run Nutch in Docker.  Check  some examples at https://github.com/sbatururimi/nutch-test.
```
You are welcome. I think it would be helpful to update the Dockerfile accordingly also (the double compilation with an for example) or the infrastructure setup using docker-selenium. What do you think about?

sbatururimi commented:
@sebastian-nagel Is everything with the pull request, anything to be changed?

sebastian-nagel commented:
Hi @sbatururimi, I've already tested it, see the comment in [NUTCH-2676](https://issues.apache.org/jira/browse/NUTCH-2676). Would be great if you could apply the code formatting, squash the commits (there are too many) and remove the duplicated README.txt. If not, please let me know and I can do it before merging. Thanks for your contribution!

sbatururimi commented:
Will do and update code before Monday. No worries, @sebastian-nagel

sbatururimi commented:
@sebastian-nagel Ready to be merged:
- duplicated readme removed
- eclipse formatting applied.

For the squashing commits, it is possible to do it right from the GitHub 
https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits

sebastian-nagel commented:
Yes, of course, I can let git/github do the squash. But writing a comprehensive commit message is still manual work. :)   I'll need 1-2 days to test the code, then I'll merge it. Thanks a lot, @sbatururimi!

sbatururimi commented:
My proposition:
```
- upgrade of Selenium plugin related packages
- added the use of headless mode when using Selenium nodes(chrome & firefox)
- obsolete code for Selenium plugin removed
- fix of a bug occurring during the build of  the Nutch docker container 
- added possibility to use a Selenium Hub orchestrator in multi-containers docker mode
- added several examples of using Nutch+Solr+Selenium Hub+Selenium Nodes in a network of Docker containers
```

sebastian-nagel commented:
Rebased, squashed and merged PR. Thanks, @sbatururimi!
